### ISSUE TITLE
Fix type_options validation for Money type

### ISSUE THREAD

**Issue by seancolsen** [centerofci/mathesar#1359]
## Reproduce

1. Set up a Money column.

1. Submit a `PATCH` request to the columns API, e.g. `/api/db/v0/tables/15/columns/52/`

1. Send:

    ```json5
    {
      "type": "MATHESAR_TYPES.MATHESAR_MONEY",
      "type_options": { "precision": 100, "scale": 2 }
    }
    ```

    Expect success.

    Receive:

    ```json
    [
      {
        "code": 4210,
        "message": "Unknown type_option passed",
        "field": null,
        "detail": null
      }
    ]
    ```

1. Send:

    ```json5
    {
      "type": "MATHESAR_TYPES.MATHESAR_MONEY",
      "type_options": { "precision": null, "scale": null }
    }
    ```

    Expect success.

    Receive:

    ```json
    [
      {
        "code": 2023,
        "field": "precision",
        "message": "This field may not be null.",
        "detail": {}
      },
      {
        "code": 2023,
        "field": "scale",
        "message": "This field may not be null.",
        "detail": {}
      }
    ]
    ```

**Comment by seancolsen** [comment]
@silentninja can you work on this?

**Comment by silentninja** [comment]
Currently, the backend does not support `type options` for `MathesarMoney`, @seancolsen I am thinking of removing `type_options` as a temporary fix, is that okay or can we wait until a proper fix.

`MathesarMoney` data type is implemented using `DOMAIN` and AFAIK `DOMAIN` does not allow passing `type options`. @mathemancer Is this correct and do we have any idea to get around it?

I can think of using default type option with `scale: 2` when creating the `DOMAIN`

**Comment by mathemancer** [comment]
> `MathesarMoney` data type is implemented using `DOMAIN` and AFAIK `DOMAIN` does not allow passing `type options`. @mathemancer Is this correct and do we have any idea to get around it?
> 
> I can think of using default type option with `scale: 2` when creating the `DOMAIN`

This is correct. If we want to do some basic type options, we'd need to create different `DOMAIN`s for each type option setting and then use the python class to init columns with the proper domain. That said, I don't think we _should_ support type options for money.

What's the use case for storing money in reduced precision/scale? I believe the standard practice for accounting is to round at the last possible moment (e.g., when displaying), and do calculations in the highest precision available.  This would imply setting the number of decimal places as a display option, not a type option. See: https://www.syspro.com/blog/applying-and-operating-erp/rounding-for-accounting-accuracy/. If they need more control, they can use a `NUMERIC` column.

**Comment by seancolsen** [comment]
Thanks for the clarification @silentninja and @mathemancer.

@mathemancer said:

> That said, I don't think we _should_ support type options for money.

I agree. I'm not sure where I got the idea that the Mathesar Money type was a `NUMERIC`. I opened this ticket because I was continuing with an assumption that I now see was probably incorrect. And I don't have a high-priority use-case in mind for rounding user-entered Money values to fit within a lower scale than their entry.

Using `display_options` seems good.

Moreover, in most cases, I would prefer to use `display_options` for our Mathesar Number type too. Our Number specs require using `type_options` to control decimal scale, which is why I've opened #1332. Within that ticket, see use-case (C) for a money-specific example that demonstrates a need for storing _two separate values_ which control the display of decimal places.

So I'm closing this ticket. I think we should continue discussion in #1332 and aim to implement `display_options` that work well for both Money and Number.

---

As an aside... From my reading of the [Postgres DOMAIN docs](https://www.postgresql.org/docs/current/domains.html), each `DOMAIN` type must have an _underlying_ type. Out of curiosity, what's the underlying Postgres type for our `MATHESAR_TYPES.MATHESAR_MONEY` type?
